src/util/json_stream.cpp

```diff
@@ -0,0 +1,142 @@
+/*******************************************************************\
+
+Module:
+
+Author: Peter Schrammel
+
+\*******************************************************************/
+
+#include "json_stream.h"
+
+#include <ostream>
+
+/// Out a delimiter to the current open JSON stream.
+void json_streamt::output_delimiter()
+{
+  if(!first)
+    out << ',';
+  else
+    first = false;
+  out << '\n';
+  out << std::string((indent+1)*2, ' ');
+}
+
+/// Construct a new JSON array stream object.
+json_stream_arrayt::json_stream_arrayt(std::ostream &out, unsigned indent)
+  : json_streamt(out, indent)
+{
+  out << '[';
+}
+
+/// Output current JSON array and close stream.
+void json_stream_arrayt::output_current()
+{
+  if(!object.empty())
+  {
+    output_delimiter();
+    object[""].output_rec(out, indent+1);
+    object.clear();
+  }
+  if(current)
+  {
+    current->close();
+  }
+}
+
+/// Output a JSON object finalizing character for a JSON array stream.
+void json_stream_arrayt::output_finalizer()
+{
+  out << '\n' << std::string(indent*2, ' ');
+  out << ']';
+}
```

Comment:
The constructor does `out << '['` so I'd expect this to be done in a destructor (otherwise the constructor shouldn't be doing the opening bracket).

Reply:
This is exactly what it is doing (by calling close() which, in turn, calls output_finalizer).